### ISSUE TITLE
Dimensões modificadas ao calcular frete

### ISSUE THREAD

**Issue by guibarbas** [claudiosanches/woocommerce-correios#182]
Ola Claudio, como vai ?

Amigo, gostaria de tirar uma dúvida sobre um problema que estamos presenciando.

Vou demonstrar abaixo com algumas imagens

1º Como esta cadastrado no produto
[https://prnt.sc/s83zo8](url)
Peso (kg): 0.012
Dimensões (cm): 7C 19L 16A

Este produto, custa R$: 2,00/un
E o pedido mínimo deve ser de 30 unidades
Para isso, estou usando o plugin (Min and Max Quantity for WooCommerce | Link: [https://br.wordpress.org/plugins/minmax-quantity-for-woocommerce/](url) )

Se puder me indicar outro que funcione melhor com seu sistema, agradeço de coração

2º Log
[https://prnt.sc/s844tg](url)
Veja como esta o log do calculo
```
04-29-2020 @ 11:55:37 - Peso e cubagem da ordem: Matriz
(
    [peso] => 0,36
    [altura] => 58
    [largura] => 19
    [comprimento] => 58
)

04-29-2020 @ 11:55:37 - Requesting Correios WebServices: http://ws.correios.com.br/calculador/CalcPrecoPrazo.aspx?nCdServico=04510&nCdEmpresa&sDsSenha&sCepDestino=14094120&sCepOrigem=11730000&nVlAltura=58&nVlLargura=19&nVlDiametro=0&nVlComprimento=58&nVlPeso=0,36&nCdFormato=1&sCdMaoPropria=N&nVlValorDeclarado=60&sCdAvisoRecebimento=N&StrRetorno=xml
04-29-2020 @ 11:55:38 - Correios WebServices response: SimpleXMLElement Object
(
    [cServico] => Objeto SimpleXMLElement
        (
            [Codigo] => 04510
            [Valor] => 53,39
            [PrazoEntrega] => 10
            [Valor sem valor adicional] => 52,60
            [ValorMaoPropria] => 0.00
            [Receipt NoticeValue] => 0.00
            [ValorValorDeclarado] => 0,79
            [EntregaDomiciliar] => S
            [EntregaSabado] => N
            [obsFim] => Objeto SimpleXMLElement
                (
                )

            [Erro] => 0
            [MsgErro] => Objeto SimpleXMLElement
                (
                )

        )

)
```

Esta somando corretamente o peso, mas modificando drasticamente as dimensões.
Isso esta impactando direto no custo.

3º Valor
[https://prnt.sc/s84571](url)

Como posso corrigir isso ?

E uma dúvida: Na aba Entrega do produto, devo inserir as dimensões do produto já na embalagem ou as dimensões reais do produto ?

**EDIÇÃO**
Claudio, Testei em outro site, que não usa o plugin de mínimo e máximo.

E obtive o mesmo problema.
Veja:

1º Config do produto
[https://prnt.sc/s84fid](url)
Peso (kg): 0.1
Dimensões (cm): 26C 15L 12A

2º Custo
[https://prnt.sc/s84g7s](url)

3º Log
[https://prnt.sc/s84du0](url)
```
04-29-2020 @ 13:28:01 - Peso e cubagem da ordem: Matriz
(
    [peso] => 1
    [altura] => 42,4
    [largura] => 42,4
    [comprimento] => 26
)

04-29-2020 @ 13:28:01 - Requesting Correios WebServices: http://ws.correios.com.br/calculador/CalcPrecoPrazo.aspx?nCdServico=04510&nCdEmpresa&sDsSenha&sCepDestino=14094120&sCepOrigem=13025320&nVlAltura=42,4&nVlLargura=42,4&nVlDiametro=0&nVlComprimento=26&nVlPeso=1&nCdFormato=1&sCdMaoPropria=N&nVlValorDeclarado=1587&sCdAvisoRecebimento=N&StrRetorno=xml
04-29-2020 @ 13:28:02 - Correios WebServices response: SimpleXMLElement Object
(
    [cServico] => Objeto SimpleXMLElement
        (
            [Codigo] => 04510
            [Valor] => 66,73
            [PrazoEntrega] => 8
            [Valor sem valor adicional] => 35,40
            [ValorMaoPropria] => 0.00
            [Receipt NoticeValue] => 0.00
            [ValorValorDeclarado] => 31.33
            [EntregaDomiciliar] => S
            [EntregaSabado] => N
            [obsFim] => Objeto SimpleXMLElement
                (
                )

            [Erro] => 0
            [MsgErro] => Objeto SimpleXMLElement
                (
                )

        )

)
```

Note que a Altura, Largura e Peso foram afetados pois selecionamos manualmente a quantidade de 7 produtos.

Aguardo seu retorno

Muito obrigado
Fique seguro.

**Comment by guibarbas** [comment]
Olá Pessoal, como estão?

Falei com um colega programador que me indicou o seguinte.

> Fiz a revisão dos códigos do plugin e está correto os cálculos da cubagem.
> 
> Segue o passo a passo de como é realizado.
> 
> 1) O plugin captura a altura x largura e comprimento de todos os itens
> 2) O plugin multiplica a altura x largura x comprimento de cada itens
> 3) O plugin após a multiplicação, soma o resultado das multiplicações
> 4) O plugin localizar a maior medida entre altura x largura x comprimento
> 5) O plugin divide o valor total das medidas pelo valor da maior medida
> 6) O plugin faz a raiz quadralada do resultado e arredonda
> 7) O plugin detecta qual das medidas é maior e atribui (no seu exemplo ele colocou 26 no log final)
> 8) O plugin atribui o resultado do cálculo da raiz quadrada como as outras medidas restantes (no seu caso 42,4)
> 9) O plugin multiplica o peso pela quantidade de itens (0,1 * 10 itens - Você informou 7 itens no post, mas os cálculos fecham com 10 itens)
> 
> O plugin sempre faz o cálculos procurando a melhor cubagem.
> 
> Abraço!

Com isso, creio que isso não seja um problema, e sim, que o sistema trabalha diferente do modo como pensamos.

Correto @claudiosanches ?

**Comment by erickamoedo** [comment]
No magento eu utilizava uma opção para ele não somar as medidas que foram adicionada ao carrinho, porém estou procurando uma solução para esse mesmo caso com Wordpress.

**Comment by claudiosanches** [comment]
@guibarbas Sim, no caso vai tentar encontrar a melhor cubagem para enviar todos os pedidos na mesma caixa, já que não calcula múltiplos pacotes ainda, algo que eu quero fazer possível um dia, é claro que com múltiplos pacotes pode ficar mais caro.
